Editorial review: Document CSS interactivity property and inert state (#38692)
mdn/content: files/en-us/web/html/reference/global_attributes/inert/index.md

sidebar: htmlsidebar
---

The **`inert`** [global attribute](/en-US/docs/Web/HTML/Reference/Global_attributes) is a boolean attribute specifying that the element and all of its flat tree descendants are inert.

Inert HTML elements and their flat tree descendants:

- Do not have {{domxref("Element/click_event", "click")}} events fired when clicked on.

> - Cannot be focused, that is, {{domxref("Element/focus_event", "focus")}} events cannot be fired on them.

Reviewer: ```suggestion
- Cannot be focused and {{domxref("Element/focus_event", "focus")}} events cannot be fired on them.
```
don't want people to think they could tab into an element, and it's just the event that doesn't happen

- Are not searchable via browser find-in-page features (none of their content is found/matched).
- Disallow users from selecting text contained within their content — akin to using the CSS property {{cssxref("user-select")}} to disable text selection.
- Cannot have otherwise-editable content edited. This includes, for example, the contents of textual {{htmlelement("input")}} fields, and text elements with [`contenteditable`](/en-US/docs/Web/HTML/Reference/Global_attributes/contenteditable) set on them.

> - Are hidden from assistive technologies by excluding them from the accessibility tree.

Reviewer: ```suggestion
- Are hidden from assistive technologies as they are excluded from the accessibility tree.
```

The following other features can be used to set an element and its descendants to an inert state:

- The CSS {{cssxref("interactivity")}} property.
- The {{domxref("HTMLElement.inert")}} DOM property.

> Modal {{htmlelement("dialog")}}s generated with [`showModal()`](/en-US/docs/Web/API/HTMLDialogElement/showModal) escape inertness, meaning that they don't inherit inertness from their ancestors, but can only be made inert by having the `inert` attribute explicitly set on themselves.

Reviewer: If the CSS property can make a dialog inert, we should got with:

 ```suggestion
Modal {{htmlelement("dialog")}}s generated with [`showModal()`](/en-US/docs/Web/API/HTMLDialogElement/showModal) escape inertness, meaning that they don't inherit inertness from their ancestors, but can be made inert by having the `inert` attribute explicitly set on themselves.
```

```html
<div inert>
  <!-- content -->
</div>
```

> The `inert` attribute can be added to sections of content that should not be interactive. When an element is inert, it — along with all of the element's descendants including normally interactive elements such as links, buttons, and form controls — are disabled because they cannot receive focus or be clicked.

Reviewer: ```suggestion
The `inert` attribute can be added to sections of content that should not be interactive. When an element is inert, it, along with all of the element's descendants, including normally interactive elements such as links, buttons, and form controls, are disabled; they cannot receive focus or be clicked.
```
Grammar 50/50: if we keep the m-dash, then we have to change "are" to "is", as the element is singular. I wasn't opposed to the em dashes, but I think it is easier and causes fewer questions to remove them here and switch to plural. (so readers don't wonder why we had a list and it was singular.... because grammar)

- HTML {{HTMLElement("dialog")}} element
- {{domxref("HTMLElement.inert")}} HTML DOM property
- CSS {{cssxref("interactivity")}} property

> - [Introducing inert](https://web.dev/articles/inert) on web.dev (2024)

Reviewer 2: I'm always slightly confused by this word choice, because our [see also guidelines](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#see_also_section) explicitly says to use "on":

> Correct: [Top-level await](https://v8.dev/features/top-level-await) on v8.dev (2019)

And I never understood why "via" here. You are not importing any content _from_ web.dev ("via" means "from" or "through"). You are taking readers _to_ web.dev.
